code/__HELPERS/time.dm

```diff
@@ -103,6 +103,35 @@ var/global/rollovercheck_last_timeofday = 0
 		return midnight_rollovers++
 	return midnight_rollovers
 
+/proc/ticks_to_text(var/ticks)
+	if(ticks%1 != 0)
+		return "ERROR"
+	var/response = ""
+	var/counter = 0
+	while(ticks >= 1 DAYS)
+		ticks -= 1 DAYS
+		counter++
+	if(counter)
+		response += "[counter] Day[counter>1 ? "s" : ""][ticks ? ", " : ""]"
+	counter=0
+	while(ticks >= 1 HOURS)
+		ticks -= 1 HOURS
+		counter++
+	if(counter)
+		response += "[counter] Hour[counter>1 ? "s" : ""][ticks?", ":""]"
+	counter=0
+	while(ticks >= 1 MINUTES)
+		ticks -= 1 MINUTES
+		counter++
+	if(counter)
+		response += "[counter] Minute[counter>1 ? "s" : ""][ticks?", ":""]"
+		counter=0
+	while(ticks >= 1 SECONDS)
+		ticks -= 1 SECONDS
+		counter++
+	if(counter)
+		response += "[counter][ticks?".[ticks]" : ""] Second[counter>1 ? "s" : ""]"
+	return response
```

Comment:
Kinda cursed. If only we had invented some sort of operation to directly extract the number of time a quantity B is present in a quantity A without looping. Maybe we could call it euclidian division or something.

Reply:
Hex only had to sacrifice three eyes of newt for this.

Reply:
Pretty sure that's a Hex line. I wonder if the downstream has this too.